tests/epyccel/test_epyccel_generators.py

```diff
@@ -199,6 +199,58 @@ def f(a0 : 'int[:]'):
 
     assert f(x) == f_epyc(x)
 
+def test_sum_with_condition(language):
+    def f():
+        v = sum(i for i in range(20) if i % 2 == 1)
+        return v
+
+    f_epyc = epyccel(f, language = language)
+    assert f() == f_epyc()
+
+def test_sum_with_multiple_conditions(language):
+    def f():
+        v = sum(i - j for i in range(20) if i % 2 == 1 for j in range(30) if j % 3 == 1)
+        return v
+
+    f_epyc = epyccel(f, language = language)
+    assert f() == f_epyc()
+
+@pytest.mark.xfail(sys.platform == "darwin", reason="Constant('inf') printing")
+def test_max_with_condition(language):
+    def f():
+        v = max(i for i in range(20) if i % 2 == 1)
+        return v
+
+    f_epyc = epyccel(f, language = language)
+    assert f() == f_epyc()
+
+@pytest.mark.xfail(sys.platform == "darwin", reason="Constant('inf') printing")
+def test_max_with_multiple_conditions(language):
+    def f():
+        v = max(i - j for i in range(20) if i % 2 == 1 for j in range(30) if j % 3 == 1)
+        return v
+
+    f_epyc = epyccel(f, language = language)
+    assert f() == f_epyc()
+
+@pytest.mark.xfail(sys.platform == "darwin", reason="Constant('inf') printing")
+def test_min_with_condition(language):
+    def f():
+        v = min(i for i in range(20) if i % 2 == 1)
+        return v
+
+    f_epyc = epyccel(f, language = language)
+    assert f() == f_epyc()
+
+@pytest.mark.xfail(sys.platform == "darwin", reason="Constant('inf') printing")
```

Comment:
Please reference an issue that describes this issue so it is easy to locate these xfails when the issue is fixed